floor/writer_test.go

```diff
@@ -281,18 +281,6 @@ func TestDecodeStruct(t *testing.T) {
 			ExpectErr:      false,
 			Schema:         `message test { required int64 ts (TIMESTAMP(NANOS, false)); }`,
 		},
-		{
```

Comment:
Marshaling/unmarshaling attributes not defined in the schema doesn't actually work at the outermost layer (write/scan), and there are a lot of failure cases for extra attribute types. To be consistent we would either need to expand this type of support to all types and move the support up a layer, or just say that we only marshal/unmarshal fields defined in the schema. The later option seems like the cleaner interface to me. To support use-cases where users don't want to have to define their schema, we can add a schema generation layer that will output a schema given an object, and that output can be passed in as the schema for marshaling.